typescript/introduction/tsc-data-types-i/tsc-any-void-null.md

```diff
@@ -0,0 +1,92 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Any, Void, Null
+
+---
+## Content
+
+### Any
+
+If we want to opt-out of type checking, we can use the `any` type:
+
+```ts
+let anythingReally: any = 1;
+anythingReally = 'one';
+anythingReally = [true, false];
+```
+
+`any` means that the value can have any type and is useful when you are unsure of the type of the data you're working with.
+
+Do not use `any` if you can apply a more specific type because allowing any type defeats the purpose of type-checking.
+
+### Void
+
+`void` is the opposite of `any`. It means that the data **shouldn't** have any type at all. 
+
+This is usually used as the return type of a function that does not return anything. 
+
+Don't worry if this is a bit confusing now, we'll explain it in future workouts. For now, know that you can only assign `undefined` as a value if the type is `void`[1]:
```

Comment:
I think we can show such a function now

```ts
function logString(input: string): void {
   console.log(input);
}
```

Reply:
the undefined assignment can be a bit confusing, I think. bc we mention there's a dedicated type for it just below

Reply:
That's a good point.

Let's just remove this paragraph and the code below it.

Reply:
Such that the whole section about `void` is just these two sentences:

```md
`void` is the opposite of `any`. It means that the data **shouldn't** have any type at all. 

This is usually used as the return type of a function that does not return anything.
```